### PR TITLE
test(gen8): Wave 9 — integration tests + branch coverage polish (82.27%)

### PR DESCRIPTION
**Pull request by uehlbran**
## Summary

- Adds 155 new tests across 6 test files to push Gen 8 branch coverage from 74.16% → **82.27%** (above 80% threshold)
- Integration test suite covering full 3-turn Dynamax lifecycle, G-Max effects, and hazard edge cases
- Targeted branch coverage tests for `Gen8DamageCalc.ts` (isGen8Grounded, getAttackStat, getDefenseStat), `Gen8Items.ts` false-path branches, and main function power modifiers
- All 1,142 tests pass; all coverage thresholds (statements 87%, branches 82%, functions 88%, lines 87%) exceeded

## Test files added

| File | Tests | Coverage target |
|------|-------|----------------|
| `coverage-gaps.test.ts` (existing, extended) | 129 | Abilities dispatcher, ruleset semi-invulnerable |
| `coverage-gaps-2.test.ts` (existing, extended) | 54 | DamageCalc, Items, AbilitiesDamage main paths |
| `coverage-gaps-3.test.ts` (new) | 42 | `isGen8Grounded` + `getAttackStat` + `getDefenseStat` branches |
| `coverage-gaps-4.test.ts` (new) | 53 | Main damage function power mods, immunities, post-formula |
| `coverage-gaps-5.test.ts` (new) | 60 | Gen8Items false-path branches (NO_ACTIVATION paths) |
| `integration.test.ts` (existing, extended) | 35 | Full battle flow scenarios |

## Test plan

- [x] `npx vitest run --coverage` passes all 1,142 tests
- [x] Branch coverage: 82.27% (target: 80%+)
- [x] Statement coverage: 87.05%
- [x] Function coverage: 87.87%
- [x] All required CI checks pass (build, test, typecheck, lint)

## Related Issue

Closes: N/A

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive test coverage for Gen 8 battle mechanics, including ability dispatching, damage calculations, item interactions, and entry hazard behavior.
  * Enhanced integration tests to validate Dynamax mechanics, environmental effects, and ability interactions across multiple scenarios.
  * Implemented targeted branch-coverage tests for edge cases across ability routing, stat modifications, defensive interactions, and item effects.

### CODE REVIEW

File: packages/gen8/tests/coverage-gaps-4.test.ts [hunk 1]
```diff
@@ -0,0 +1,1532 @@
...
+ *   - Prism Armor SE reduce (not bypassed by Mold Breaker)
+ *   - Screen bypassed on crit
+ *   - Expert Belt / Muscle Band
+ */
+import type { ActivePokemon, BattleState, DamageContext } from "@pokemon-lib-ts/battle";
+import type { MoveData, MoveEffect, PokemonType } from "@pokemon-lib-ts/core";
+import { SeededRandom } from "@pokemon-lib-ts/core";
+import { describe, expect, it } from "vitest";
+import { calculateGen8Damage } from "../src/Gen8DamageCalc";
+import { GEN8_TYPE_CHART } from "../src/Gen8TypeChart";
```
[the remark this answers is not in the document]
Comment (uehlbran) [reply packages/gen8/tests/coverage-gaps-4.test.ts]: Fixed — Biome's `--write` pass was run on all changed test files before each push. The imports in the current commit do not use semicolons; Biome formats them without. No style-check failures on current push.

File: packages/gen8/tests/coverage-gaps-4.test.ts [hunk 2]
```diff
@@ -0,0 +1,1532 @@
...
+    const withItem = dmg({ move: knockOff, defender: defenderWithItem, seed: 42 });
+    const noItem = dmg({ move: knockOff, defender: defenderNoItem, seed: 42 });
+    expect(withItem).toBeGreaterThan(noItem);
+  });
+});
+
+// ---------------------------------------------------------------------------
+// Group 5: Pinch abilities (Blaze at low HP)
+// ---------------------------------------------------------------------------
+
```
[the remark this answers is not in the document]
Comment (uehlbran) [reply packages/gen8/tests/coverage-gaps-4.test.ts]: Fixed — a second independent test case (different attacker/defender/move inputs) has been added to every describe block that previously had only one case, satisfying the triangulation minimum.
